hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -213,7 +213,9 @@ private OzoneBucket getBucket(String volumeStr, String bucketStr,
     try {
       bucket = proxy.getBucketDetails(volumeStr, bucketStr);
     } catch (OMException ex) {
-      if (createIfNotExist) {
+      // Note: always create bucket if volumeStr matches "tmp" so -put works
+      if (createIfNotExist ||
+          volumeStr.equals(OFSPath.OFS_MOUNT_TMP_VOLUMENAME)) {
```

Comment:
Do we create the tmp bucket even for read operation?

Reply:
That's correct. This is my implementation decision.

I initially only create temp bucket for write, as you would expect. Then I found Hadoop FS file write operations calls `getFileStatus` beforehand.

Reply:
After some discussion, we decide to let user create their temp bucket manually upon first use. i.e. `ozone fs -mkdir ofs://om/tmp/`

Reply:
Addressed in 1d4164b153386c5e476f27df6420512b60894d5d